content/300-guides/050-database/820-multi-schema.mdx

```diff
@@ -0,0 +1,101 @@
+
+</TopBlock>
+
+## What are database schemas?
+
+Many database providers allow you to create named collections within a database that group together tables, constraints, indexes and other database features. In PostgreSQL and SQL Server these collections are known as schemas. We will refer to these collections as _database schemas_ to distinguish them from Prisma's own schema file.
+
+## How to enable the <inlinecode>multiSchema</inlinecode> preview feature
+
+Multi-schema support is currently in an early, experimental preview. To enable the `multiSchema` preview feature, you will need to add the `multiSchema` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema file:
+
+```prisma file=schema.prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["multiSchema"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+## How to include multiple database schemas in your Prisma schema
+
+To use multiple database schemas in your Prisma schema file, add the names of your database schemas to an array in a new `schemas` field of the `datasource` block. The following example adds a `"base"` and a `"transactional"` schema:
+
+```prisma file=schema.prisma highlight=9;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["multiSchema"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+  schemas  = ["base", "transactional"]
+}
+```
+
+You do not need to change your connection string. You can still use the `schema` key of your connection string to refer to the default schema, which will be used for Prisma Client raw queries.
+
+To mark a model as belonging to a specific database schema, add a `@@schema` attribute with the name of the database schema as a parameter. In the following example the `User` model is part of the `"base"` schema, and the `Order` model is part of the `"transactional"` schema:
+
+```prisma file=schema.prisma highlight=6,14;add
+model User {
+  id     Int     @id
+  orders Order[]
+
+  @@schema("base")
+}
+
+model Order {
+  id      Int  @id
+  user    User @relation(fields: [id], references: [id])
+  user_id Int
+
+  @@schema("transactional")
+}
+```
+
+<Admonition type="warning">
+
+It is currently not allowed to have models with the same name in different schemas. This will fail to validate in your Prisma schema.
+
+</Admonition>
+
+## How to query across multiple schemas with Prisma Client
+
+You can query models in multiple database schemas without any change to your Prisma Client query syntax. For example, the following query finds all orders for a given user using the Prisma schema above:
+
+```ts
+const orders = await prisma.order.findMany({
+  where: {
+    user: {
+      id: 1,
+    },
+  },
+})
+```
```

Comment:
Feels like we need a proper ending. Even though I am not a fan of linking to temporal issues in our docs, perhaps we should add a sentence like "To learn more about future developments in this preview feature, refer to Issue #____" and then link to the feedback issue for this Preview feature?

Reply:
Yes agree that that's better, and the issue should be included anyway - I've added that in in a final section.